ddtrace/tracer/sqlcomment.go

```diff
@@ -70,6 +71,38 @@ type SQLCommentCarrier struct {
 	Mode          DBMPropagationMode
 	DBServiceName string
 	SpanID        uint64
+	Tags          map[string]interface{}
+}
+
+func (c *SQLCommentCarrier) getMeta(k string) (string, bool) {
+	v, ok := c.Tags[k]
+	if !ok {
+		return "", false
+	}
+	if s, ok := v.(string); ok {
+		return s, true
+	}
+	return "", false
+}
+
+func (c *SQLCommentCarrier) getDBService() string {
+	dbService := c.DBServiceName
+
+	tr, ok := internal.GetGlobalTracer().(*tracer)
+	if !ok {
+		return dbService
+	}
+	peerService, _ := c.getMeta(ext.PeerService)
+	if peerService != "" {
+		return peerService
+	}
+	if !tr.config.peerServiceDefaultsEnabled {
+		return dbService
+	}
+	if peerService, _ := getPeerServiceFromSource(c); peerService != "" {
```

Comment:
I ran go test with coverage, this function is only covered by tests lines 89 to 94. I suggest writing a small test, or adding more test cases in `TestSQLCommentCarrier`.